src/langsmith/assistants.mdx

```diff
@@ -3,36 +3,179 @@ title: Assistants
+    style B fill:#E8F4F8,stroke:#4A90E2,stroke-width:2px
+    style C fill:#E8F4F8,stroke:#4A90E2,stroke-width:2px
+    style D fill:#E8F4F8,stroke:#4A90E2,stroke-width:2px
+```
+
+- **Environment-specific configurations**
+  - Use different models or settings for development, staging, and production.
+  - Test configuration changes in staging before promoting to production.
+  - Reduce costs in non-production environments with smaller models.
+
+- **A/B testing and experimentation**
+  - Compare different prompts, models, or parameter settings.
+  - Roll out configuration changes gradually to a subset of users.
+  - Measure performance differences between configuration variants.
+
+- **Specialized task variants**
+  - Create domain-specific versions of a general-purpose agent.
+  - Optimize configurations for different languages, regions, or industries.
+  - Maintain consistent graph logic while varying the execution details.
+
+```mermaid
+graph TD
+    A["Graph: writing-agent<br/>(deployed)"]
+    A --> B["Blog Assistant<br/>━━━━━━━━━━━━━<br/>Model: GPT-4<br/>Tone: Formal<br/>Style: Long-form<br/>Tools: SEO optimization"]
+    A --> C["Tweet Assistant<br/>━━━━━━━━━━━━━<br/>Model: GPT-4-mini<br/>Tone: Casual<br/>Style: 280-char limit<br/>Tools: Hashtag suggestions"]
+    A --> D["Email Assistant<br/>━━━━━━━━━━━━━<br/>Model: GPT-4<br/>Tone: Professional<br/>Style: Medium length<br/>Tools: Templates"]
+
+    style A fill:#4A90E2,stroke:#2E5C8A,stroke-width:3px,color:#fff
+    style B fill:#E8F4F8,stroke:#4A90E2,stroke-width:2px
+    style C fill:#E8F4F8,stroke:#4A90E2,stroke-width:2px
+    style D fill:#E8F4F8,stroke:#4A90E2,stroke-width:2px
+```
+
+## How assistants work with deployments
+
+When you deploy a graph with LangSmith Deployment, [Agent Server](/langsmith/agent-server) automatically creates a **default assistant** tied to that graph's default configuration. You can then create additional assistants for the same graph, each with its own configuration.
+
+Assistants have several key features:
+
+- **[Managed via API and UI](/langsmith/configuration-cloud)**: Create, list, update, version, and select assistants using the Agent Server/LangGraph SDKs or the [LangSmith UI](https://smith.langchain.com).
+- **One graph, multiple assistants**: A single deployed graph can support multiple assistants, each with different configurations (e.g., prompts, models, tools).
+- **[Versioned](#versioning) configurations**: Each assistant maintains its own configuration history through versioning. Editing an assistant creates a new version, and you can promote or roll back to any version.
+- **[Configuration](#configuration) updates without graph changes**: Update prompts, model selection, and other settings through assistant configurations, enabling rapid iteration without modifying or redeploying your graph code.
+
+<Note>
+When invoking a run, you can specify either:
+- A **graph name** (e.g., `"agent"`): Uses the default assistant for that graph
```

Comment:
Suggestion: We should mention where the `graph_name` is specified. It's in `langgraph.json`.
```json
{
    "graphs": {
        "graph_name_1": "path_to_graph_id_1",  // <-- default assistant is created for graph_id_1
        "graph_name_2": "path_to_graph_id_2"  // <-- default assistant is created for graph_id_2
    }
}
```

Separately, let's get consistent with `graph ID` and `graph name` (everywhere in this PR). Either one is fine to me, but we call it `Graph ID` in the LangGraph CLI reference page. It's also `graph_id` internally in our code.